src/platforms/macos/daemon/dnsutilsmacos.cpp

```diff
@@ -19,205 +17,52 @@ Logger logger("DnsUtilsMacos");
-                                   CFSTR("mozillavpn"), nullptr, nullptr);
-  if (m_scStore == nullptr) {
-    logger.error() << "Failed to create system configuration store ref";
-  }
+  m_dnsManager.setProcessChannelMode(QProcess::MergedChannels);
+  connect(&m_dnsManager, SIGNAL(readyReadStandardOutput()), this,
+          SLOT(dnsManagerStdoutReady()));
 
   logger.debug() << "DnsUtilsMacos created.";
 }
 
 DnsUtilsMacos::~DnsUtilsMacos() {
   MZ_COUNT_DTOR(DnsUtilsMacos);
-  restoreResolvers();
-  logger.debug() << "DnsUtilsMacos destroyed.";
-}
-
-static QString cfParseString(CFTypeRef ref) {
-  if (CFGetTypeID(ref) != CFStringGetTypeID()) {
-    return QString();
-  }
-
-  CFStringRef stringref = (CFStringRef)ref;
-  CFRange range;
-  range.location = 0;
-  range.length = CFStringGetLength(stringref);
-  if (range.length <= 0) {
-    return QString();
-  }
-
-  UniChar* buf = (UniChar*)malloc(range.length * sizeof(UniChar));
-  if (!buf) {
-    return QString();
-  }
-  auto guard = qScopeGuard([&] { free(buf); });
-
-  CFStringGetCharacters(stringref, range, buf);
-  return QString::fromUtf16(buf, range.length);
-}
-
-static QStringList cfParseStringList(CFTypeRef ref) {
-  if (CFGetTypeID(ref) != CFArrayGetTypeID()) {
-    return QStringList();
-  }
-
-  CFArrayRef array = (CFArrayRef)ref;
-  QStringList result;
-  for (CFIndex i = 0; i < CFArrayGetCount(array); i++) {
-    CFTypeRef value = CFArrayGetValueAtIndex(array, i);
-    result.append(cfParseString(value));
+  if (!restoreResolvers()) {
+    m_dnsManager.kill();
   }
-
-  return result;
-}
-
-static void cfDictSetString(CFMutableDictionaryRef dict, CFStringRef name,
-                            const QString& value) {
-  if (value.isNull()) {
-    return;
-  }
-  CFStringRef cfValue = CFStringCreateWithCString(
-      kCFAllocatorSystemDefault, qUtf8Printable(value), kCFStringEncodingUTF8);
-  CFDictionarySetValue(dict, name, cfValue);
-  CFRelease(cfValue);
+  logger.debug() << "DnsUtilsMacos destroyed.";
 }
 
-static void cfDictSetStringList(CFMutableDictionaryRef dict, CFStringRef name,
-                                const QStringList& valueList) {
-  if (valueList.isEmpty()) {
-    return;
-  }
-
-  CFMutableArrayRef array;
-  array = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0,
-                               &kCFTypeArrayCallBacks);
-  if (array == nullptr) {
-    return;
-  }
-
-  for (const QString& rstring : valueList) {
-    CFStringRef cfAddr = CFStringCreateWithCString(kCFAllocatorSystemDefault,
-                                                   qUtf8Printable(rstring),
-                                                   kCFStringEncodingUTF8);
-    CFArrayAppendValue(array, cfAddr);
-    CFRelease(cfAddr);
+void DnsUtilsMacos::dnsManagerStdoutReady() {
+  for (;;) {
+    QByteArray line = m_dnsManager.readLine();
+    if (line.length() <= 0) {
+      break;
+    }
+    logger.debug() << QString::fromUtf8(line);
```

Comment:
Should this add a string like "DnsManagerStdout" so it is easier to find it in a log?

Reply:
Done!